### DISCUSSION TITLE
Receiving sensor messages in incorrect order?

### DISCUSSION THREAD

**Discussion by rsilves** [arendst/Tasmota#18995] [Support request] [answered]
Hi everyone.
I'm trying to configure a Philips Hue Filament bulb with my Sonoff Zigbee Bridge Pro, flashed with Tasmota and connected to Home Assistant via MQTT.

The issue I'm having is that every time I send an OFF command to the lamp ( {"Power": 0} ) it first responds with a message like this:
`{"ZbReceived":{"Globe_Light":{"Device":"0xEB8C","Name":"Globe_Light","Power":0,"Dimmer":58,"Endpoint":11,"LinkQuality":145}}}
`
(Note that it says "Power:0"

And then it sends a new message like this:

`{"ZbReceived":{"Globe_Light":{"Device":"0xEB8C","Name":"Globe_Light","Dimmer":184,"Endpoint":11,"LinkQuality":145}}}
`
(Note that it doesn't mention "Power" and it shows a higher Dimmer value) than before).

I believe this happens because the messages are not being sent in the correct order.

I would appreciate any recommendations here. I read that there are commands for specifying the transition time but I don't know how to send those to the lamp or the hub.. Anyway it would be ideal to know of a way to make the Power: 0 message arrive last after the light is actually off, or avoid the intermediate messages.

Thanks,
Rodrigo

**Comment by Jason2866** [comment]
Probably HA is to blame. Zigbee devices just receives what has been sent.

**Reply by rsilves** [reply]
Hi @Jason2866. thanks for your reply. I'm monitoring the MQTT messages received through the MQTT server's "listen to a topic" tool. They're yet to be interpreted by HA.

This same bulb was configured before with this same Zigbee Bridge Pro, but it was connected to HA through ZHA over TCP. I cannot use that configuration now. I can either use Zigbee2MQTT (I wasn't successful when trying to set it up with this hub over TCP) or I can use the MQTT messages to create an entity in Home Assistant.

But this issue happens way before creating the entity, I'm seeing the problem looking at the messages in the MQTT server. Do you think the server can be the problem here?

Thanks,
Rodrigo

EDIT:
Same thing happens when looking at the Tasmota web console:
`20:04:22.372 MQT: stat/ZBBridgePro/RESULT = {"ZbSend":"Done"}
20:04:23.217 MQT: tele/ZBBridgePro/SENSOR = {"ZbReceived":{"Globe_Light":{"Device":"0xEB8C","Name":"Globe_Light","Power":0,"Dimmer":190,"Endpoint":11,"LinkQuality":149}}}
20:04:24.016 MQT: tele/ZBBridgePro/SENSOR = {"ZbReceived":{"Globe_Light":{"Device":"0xEB8C","Name":"Globe_Light","Dimmer":237,"Endpoint":11,"LinkQuality":149}}}`

**Comment by Jason2866** [comment]
No, mqtt is a broker. It offers Infos. Not the same as sending. What command is/are published (for the bulb)?
When the mqtt broker gets more commands (in wrong order) it is initiated from "somewhere".
This "somewhere" is doing not like you want.
There are no zigbee data telegrams sent without an mqtt command to do so!

**Reply by rsilves** [reply]
The command published is
`{"device":"Globe_Light","send":{"Power": 0}}`
Edit:
(This is the command I publish to get the light off)

**Comment by Jason2866** [comment]
So, if no other command is published, the bulb itself does the weird dimmer change.

**Comment by rsilves** [comment, marked as answer]
It seems I found the solution:

When I send the Power:0 command, this mixed messages issue happens. But if I send Dimmer:0 instead, then the last message received says Power:0 and it all works fine.

**Reply by s-hadinger** [reply]
This is probably due to the way the bulb handles messages
